analyzers/src/SonarAnalyzer.CSharp/Rules/RedundantJumpStatement.cs

```diff
@@ -148,7 +148,7 @@ private static bool IsOnlyYieldBreak(JumpBlock jumpBlock, int yieldStatementCoun
 
         private static bool IsThrow(JumpBlock jumpBlock)
         {
-            return jumpBlock.JumpNode is ThrowStatementSyntax;
+            return jumpBlock.JumpNode is ThrowStatementSyntax || ThrowExpressionSyntaxWrapper.IsInstance(jumpBlock.JumpNode);
```

Comment:
More lightweight solution:
```suggestion
            return jumpBlock.JumpNode.IsAnyKind(SyntaxKind.ThrowStatement, SyntaxKindEx.ThrowExpression);
```